pokemon-api/src/main/java/cz/muni/fi/pa165/pokemon/league/participation/manager/facade/BadgeFacade.java

```diff
@@ -0,0 +1,32 @@
+package cz.muni.fi.pa165.pokemon.league.participation.manager.facade;
+
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.BadgeCreateDTO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.BadgeDTO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.BadgeStatusChangeDTO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.GymDTO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.TrainerDTO;
+
+/**
+ * Interface for Badge Facade
+ *
+ * @author Michal Mokros 456442
+ */
+public interface BadgeFacade {
+
+    Long createBadge(BadgeCreateDTO badge);
+
+    BadgeDTO findBadgeById(Long id);
+
+    GymDTO getGymOfBadge(Long id);
+
+    TrainerDTO getTrainerOfBadge(Long id);
+
+    void revokeBadgeForTrainer(Long trainerId, BadgeStatusChangeDTO badge);
```

Comment:
Which trainer is which? Anyway, you only need the trainer that requested the change (usually the gym leader, except for challenge trainer, which seems unnecessary anyway), you can get the challenging trainer from the badge if needed.